src/backend.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+let debug = require('debug')(require('path').relative(process.cwd(), __filename));
+let aws = require('aws-sdk');
+let sqs = new aws.SQS({apiVersion: '2012-11-05', region: 'us-west-2'});
+let awsPromise = require('aws-sdk-promise');
+let Memcached = require('memcache-promise');
+let storageBackend = require('./storage-backend');
+let memcached = new Memcached('localhost:11211');
+//require('source-map-support/register')
+
+let setupBackends = async function (config) {
+  let awsBackends = [];
+  let awsRegions = config.awsRegions;
+
+  for (let region of awsRegions) {
```

Comment:
This is a not, but you would get better concurrency with:
```js
let backends = await Promise.all(awsRegions.map(async (region) => {
  let s3 = new awsPromise.s3(...);
  let bucket = config.s3BucketBase + region;
  await s3.createBucket(...).catch(err => {
    if (err.code != ...) {
      throw err;
    }
  });

  let backend = ....
  await backend.init();
  return backend;
}));
```

Note, just illustrative... But creating S3 buckets can take a while doing it sequentially is not ideal.
The `.catch(...` things is just another way of catching async errors, I use it sometimes when it feels more compact, but try/catch are perfectly fine too.